frontend/model/captureLogs.js

```diff
@@ -220,20 +164,41 @@ export function downloadLogs (elLink: Object): void {
   }
 }
 
-function clearLogs () {
-  let i = localStorage.length
-  while (i--) {
-    const key = localStorage.key(i)
-    if (typeof key === 'string' && key.indexOf('giConsole') >= 0) {
-      localStorage.removeItem(key)
+export function getLogger (): Object {
+  if (!logger) {
+    logger = createLogger(config)
+    const previousEntries = JSON.parse(getItem('entries'))
+
+    // If `maxEntries` is changed in a release, this will discard oldest logs as necessary.
+    if (config.maxEntries < previousEntries?.length) {
+      previousEntries.splice(0, previousEntries.length - config.maxEntries)
+    }
+    // Load the previous entries to sync the in-memory array with the local storage.
+    if (previousEntries?.length) {
+      logger.entries.addAll(previousEntries)
     }
   }
-  lastEntry = ''
-  entriesCount = 0
+  return logger
 }
 
-function resetLogs () {
-  clearLogs()
-  // make sure all configs are set correctly
-  verifyLogsConfig()
+function setAppLogsFilter (filter: Array<string>) {
+  appLogsFilter = filter
+  // NOTE: Find a way to capture logs without messing up with log file location.
+  // console.log() doesnt include stack trace, so when logged, we can't access
+  // where the log came from (file name), which} difficults debugging if needed.
+  for (const methodName of appLogsFilter) {
+    console[methodName] = logger[methodName]
+    if (console[methodName] !== logger[methodName]) {
+      throw new Error(`could not overwrite console method ${methodName}`)
+    }
+  }
```

Comment:
If a checkbox for a log level is checked in the UI, that logging should appear in both the console and in the UI.

And if a checkbox for a log level is unchecked, then it shouldn't appear in either.

I think there's a bug here because while it enables logging, I don't think it disables logging.

Reply:
That's true. Unchecked log levels aren't elements of `appLogsFilter` and therefore the corresponding console methods currently don't get overwritten.

This can be fixed if we only consider `debug`, `error`, `info`, `log` and `warn`, but please note that the console object has yet other methods which still log something if called. Although we currently don't use any of them I think.

Reply:
> but please note that the console object has yet other methods which still log something if called. Although we currently don't use any of them I think.

Totally OK. We'll just have to avoid using those other ones.